internal/utils/utils.go

```diff
@@ -469,7 +469,7 @@ func GetProxyEnvVars() []v1.EnvVar {
 				}
 			}
 			envVars = append(envVars, v1.EnvVar{
-				Name:  envvar,
+				Name:  strings.ToLower(envvar),
```

Comment:
This implies we are now setting the same env var twice which is unnecessary.  These vars come from the global proxy adding them to the operator.  Do we know if they ALWAYS populate the lower case version?  Is it possible to simply remove the uppercase version from the list we check?

Reply:
The `cluster-logging-operator` container has these variables exported in uppercase only, so won't be any duplication:
```
$ oc get proxy/cluster -o yaml | yq '.spec'
httpProxy: http://username:password@X.X.X.X:8080
httpsProxy: http://username:password@X.X.X.X:8080
trustedCA:
  name: ""

$ oc get po -n openshift-logging                                                                            
NAME                                        READY   STATUS    RESTARTS   AGE
cluster-logging-operator-56b476f846-qxxzj   1/1     Running   0          47s

$ oc -n openshift-logging exec -ti cluster-logging-operator-56b476f846-qxxzj -- sh -c 'env | egrep -i proxy'
HTTP_PROXY=http://username:password@X.X.X.X:8080
NO_PROXY=.cluster.local,.eu-central-1.compute.internal,.svc,10.0.0.0/16,10.128.0.0/14,127.0.0.1,169.254.169.254,172.30.0.0/16,api-int.gmeghnag.emeashift.support,localhost
HTTPS_PROXY=http://username:password@X.X.X.X:8080

$ oc get csv -n openshift-logging
NAME                                     DISPLAY                            VERSION   REPLACES                                 PHASE
cluster-logging.5.5.5                    Red Hat OpenShift Logging          5.5.5                                              Succeeded
elasticsearch-operator.5.5.5             OpenShift Elasticsearch Operator   5.5.5                                              Succeeded
```

Reply:
This still may not be the correct solution given this function is likely also used to deploy the vector container as the collector.  Does vector honor these lower case vars?  Does it expect uppercase?  There is more work to be done to ensure the functionality remains with no regression

Reply:
Right points to consider, I will inquire about them:
- How does [vector](https://github.com/vectordotdev/vector) get the proxy URL to use?
- If it takes the proxy URL from the environment variables does it support these variables to be in lowercase?

Reply:
@gmeghnag are these questions directed at me because they read like the questions I asked of you

Reply:
@jcantrill No, the questions are just notes to remind me what information I need to get, I will try by the end of the week to get information on that.

Reply:
from [vector](https://github.com/vectordotdev/vector) issue [#6401](https://github.com/vectordotdev/vector/issues/6401#:~:text=Additionally%20we%20should%20read%20from%20the%20following%20env%20vars:), pull [#8205](https://github.com/vectordotdev/vector/pull/8205#discussion_r678568354) both the lowercase and uppercase form are supported.

Reply:
@gmeghnag looks like we need a fix of unit tests and then we are ready to go

Reply:
Opened https://github.com/openshift/cluster-logging-operator/pull/1875 to fix unit tests.